internal/pkg/controller/controller_test.go

```diff
@@ -0,0 +1,239 @@
+package controller
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Reloader/pkg/kube"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	client, _           = kube.GetClient()
+	configmapNamePrefix = "testconfigmap-reloader"
+	secretNamePrefix    = "testsecret-reloader"
+	letters             = []rune("abcdefghijklmnopqrstuvwxyz")
+)
+
+func randSeq(n int) string {
+	rand.Seed(time.Now().UnixNano())
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+// Creating a Controller to do a rolling upgrade upon updating the configmap or secret
+func TestControllerForUpdatingConfigmapShouldUpdateDeployment(t *testing.T) {
+	namespace := "test-reloader"
+	createNamespace(t, namespace)
+	defer deleteNamespace(t, namespace)
+
+	controller, err := NewController(client, "configMaps", namespace)
+	if err != nil {
+		logrus.Infof("Unable to create NewController error = %v", err)
+		return
+	}
+	stop := make(chan struct{})
+	defer close(stop)
+	go controller.Run(1, stop)
+	time.Sleep(10 * time.Second)
+
+	configmapName := configmapNamePrefix + "-update-" + randSeq(5)
+	configmapClient := client.CoreV1().ConfigMaps(namespace)
+	_, err = configmapClient.Create(initConfigmap(namespace, configmapName))
+	if err != nil {
+		logrus.Infof("Error detected %s.\n", err)
+		panic(err)
+	}
+	logrus.Infof("Created Configmap %q.\n", configmapName)
+	time.Sleep(10 * time.Second)
+	deployment := createDeployement(configmapName, namespace)
+
+	logrus.Infof("Updating Configmap %q.\n", configmapName)
+	_, err = configmapClient.Get(configmapName, metav1.GetOptions{})
+	if err != nil {
+		logrus.Infof("Error while getting configmap %v", err)
+	}
+	_, updateErr := configmapClient.Update(updateConfigmap(namespace, configmapName))
+
+	// TODO: Add functionality to verify reloader functionality here
+
+	if updateErr != nil {
+		error := controller.client.CoreV1().ConfigMaps(namespace).Delete(configmapName, &metav1.DeleteOptions{})
+		logrus.Infof("Error while deleting the configmap %v", error)
+		panic(updateErr)
+	}
+	time.Sleep(10 * time.Second)
+	logrus.Infof("Deleting Deployment %q.\n", deployment.GetObjectMeta().GetName())
+	controller.client.Extensions().Deployments(namespace).Delete(configmapName, &metav1.DeleteOptions{})
```

Comment:
Error return value of `(github.com/stakater/Reloader/vendor/k8s.io/client-go/kubernetes/typed/extensions/v1beta1.DeploymentInterface).Delete` is not checked